src/script/src/python/pyo3/vector_impl.rs

```
@@ -0,0 +1,277 @@
+        if pyo3_is_obj_scalar(other.as_ref(py)) {
+            self.pyo3_scalar_arith_op(py, other, None, wrap_result(arithmetic::subtract_dyn))
+        } else {
+            self.pyo3_vector_arith_op(py, other, None, wrap_result(arithmetic::subtract_dyn))
+        }
+    }
+    fn __rsub__(&self, py: Python<'_>, other: PyObject) -> PyResult<Self> {
+        if pyo3_is_obj_scalar(other.as_ref(py)) {
+            self.pyo3_scalar_arith_op(
+                py,
+                other,
+                None,
+                wrap_result(|a, b| arithmetic::subtract_dyn(b, a)),
+            )
+        } else {
+            self.pyo3_vector_arith_op(
+                py,
+                other,
+                None,
+                wrap_result(|a, b| arithmetic::subtract_dyn(b, a)),
+            )
+        }
+    }
+    fn __mul__(&self, py: Python<'_>, other: PyObject) -> PyResult<Self> {
+        if pyo3_is_obj_scalar(other.as_ref(py)) {
+            self.pyo3_scalar_arith_op(py, other, None, wrap_result(arithmetic::multiply_dyn))
+        } else {
+            self.pyo3_vector_arith_op(py, other, None, wrap_result(arithmetic::multiply_dyn))
+        }
+    }
+    fn __rmul__(&self, py: Python<'_>, other: PyObject) -> PyResult<Self> {
+        self.__mul__(py, other)
+    }
+    fn __truediv__(&self, py: Python<'_>, other: PyObject) -> PyResult<Self> {
+        if pyo3_is_obj_scalar(other.as_ref(py)) {
+            self.pyo3_scalar_arith_op(
+                py,
+                other,
+                Some(ArrowDataType::Float64),
+                wrap_result(arithmetic::divide_dyn),
+            )
+        } else {
+            self.pyo3_vector_arith_op(
+                py,
+                other,
+                Some(ArrowDataType::Float64),
+                wrap_result(arithmetic::divide_dyn),
+            )
+        }
+    }
+    #[allow(unused)]
+    fn __rtruediv__(&self, py: Python<'_>, other: PyObject) -> PyResult<Self> {
+        todo!()
```

Better to return an error, `todo!` will panic.